e2e-tests/path-prefix/scripts/serve.js

```diff
@@ -1,22 +1,9 @@
-const handler = require(`serve-handler`)
 const http = require(`http`)
-const path = require(`path`)
+const httpProxy = require("http-proxy")
 
+const proxy = httpProxy.createProxyServer()
 const server = http.createServer((request, response) =>
-  handler(request, response, {
-    public: path.resolve(`assets`),
-    headers: [
-      {
-        source: `**/*`,
-        headers: [
-          {
-            key: `Access-Control-Allow-Origin`,
-            value: `*`,
-          },
-        ],
-      },
-    ],
-  })
+  proxy.web(request, response, { target: "http://localhost:9000" })
```

Comment:
used proxy here instead of separate serve because of "on-demand" behaviour of DSG/SSR which would need to be either replicated here (or just forward request to serve to let it handle it)